### PR TITLE
feat: refactor new error types (campaign_manager, candu, canny, clevertap, confluent_cloud)

### PR DESCRIPTION
**Pull request by ujjwal-ab**
## Description of the change

- Campaign Manager
- Candu
- Canny
- Clevertap
- Confluent Cloud

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: src/v0/destinations/campaign_manager/networkHandler.js [hunk 1]
```diff
@@ -74,15 +78,22 @@ const responseHandler = destinationResponse => {
       destinationResponse
     };
   }
+  if (getAuthErrCategory(status) === "REFRESH_TOKEN") {
+    throw new RetryableError(
```
Comment (saikumarrs) [inline src/v0/destinations/campaign_manager/networkHandler.js:82]: This should be `InvalidAuthTokenError` error.

Comment (saikumarrs) [reply src/v0/destinations/campaign_manager/networkHandler.js:82]: and also set the `authErrCategory`.

File: src/v0/destinations/canny/util.js [hunk 2]
```diff
@@ -40,16 +44,17 @@ const retrieveUserId = async (apiKey, message) => {
...
+      throw new NetworkInstrumentationError(
+        `[Canny]:: CannyUserID can't be gnerated due to ${response.data.error}`
       );
     }
     return (
       response?.response?.data?.data?.id || response?.response?.data?.id || null
     );
   } catch (error) {
-    throw new CustomError("Unable to retrieve userid from Canny", 400);
+    throw new NetworkInstrumentationError(
```
Comment (saikumarrs) [inline src/v0/destinations/canny/util.js:55]: This should be `TransformationError`.

File: src/v0/destinations/canny/util.js [hunk 3]
```diff
@@ -40,16 +44,17 @@ const retrieveUserId = async (apiKey, message) => {
     response = await httpPOST(url, qs.stringify(requestBody), header);
     // If the request fails, throwing error.
     if (response.success === false) {
-      throw new CustomError(
-        `[Canny]:: CannyUserID can't be gnerated due to ${response.data.error}`,
-        400
+      throw new NetworkInstrumentationError(
```
Comment (saikumarrs) [inline src/v0/destinations/canny/util.js:47]: How can we deterministically call it an instrumentation error?

**Review by saikumarrs — CHANGES_REQUESTED** [review]

File: src/v0/destinations/canny/util.js (hunk 3 above)
Comment (ujjwal-ab) [reply src/v0/destinations/canny/util.js:47]: Here, the request is failing but we are not sure about the reason. But we are taking all the fields for the request body from the user only. What should be the suitable error category for this?

File: src/v0/destinations/campaign_manager/networkHandler.js [hunk 4]
```diff
@@ -74,15 +79,23 @@ const responseHandler = destinationResponse => {
...
-    .setAuthErrorCategory(getAuthErrCategory(status))
-    .build();
+  throw new NetworkError(
+    `Campaign Manager: ${response.error.message} during CAMPAIGN_MANAGER response transformation 3`,
+    status,
+    {
+      [tags.TAG_NAMES.ERROR_TYPE]: getDynamicErrorType(status)
+    },
+    destinationResponse
+  );
```
Comment (utsabc) [inline src/v0/destinations/campaign_manager/networkHandler.js:98]: Please add auth error category

File: src/v0/destinations/canny/util.js (hunk 3 above)
Comment (utsabc) [reply src/v0/destinations/canny/util.js:48]: This should be safely attributed to NetworkError in order to fwd details to the caller service

Comment (saikumarrs) [reply src/v0/destinations/canny/util.js:48]: Yes.

**Review by utsabc — APPROVED** [review]

**Pull request merged**